src/appleseed/renderer/modeling/scene/scene.cpp

```diff
@@ -209,6 +209,50 @@ namespace
 
         return false;
     }
+
+    bool assembly_instances_has_participating_media(
+        const AssemblyInstanceContainer&  assembly_instances,
+        set<UniqueID>&                    visited_assemblies)
+    {
+        // Regarding participating media in the Tracer,
+        // we only care about camera and shadow rays.
+        const uint32 visibility_mask = VisibilityFlags::CameraRay | VisibilityFlags::ShadowRay;
```

Comment:
Best use `VisibilityFlags::Type` instead of `uint32` (same thing in the method above).